src/Support/Reports/Volumes/ImageAnnotations/CocoReportGenerator.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+namespace Biigle\Modules\Reports\Support\Reports\Volumes\ImageAnnotations;
+
+use Biigle\LabelTree;
+use Biigle\Modules\Reports\Support\CsvFile;
+use Biigle\User;
+
+use Biigle\Modules\Reports\Support\Reports\MakesZipArchives;
+use DB;
+
+class CocoReportGenerator extends AnnotationReportGenerator
+{
+    use MakesZipArchives;
+    /**
+     * Name of the report for use in text.
+     *
+     * @var string
+     */
+    public $name = 'coco image annotation report';
+
+    /**
+     * Name of the report for use as (part of) a filename.
+     *
+     * @var string
+     */
+    public $filename = 'coco_image_annotation_report';
+
+    /**
+     * File extension of the report file.
+     *
+     * @var string
+     */
+    public $extension = 'zip';
+
+    /**
+     * Generate the report.
+     *
+     * @param string $path Path to the report file that should be generated
+     */
+    public function generateReport($path)
+    {
+        //restrictions 
+        $rows = $this->query()->get();
+        $toZip = [];
+
+        if ($this->shouldSeparateLabelTrees() && $rows->isNotEmpty()) {
+            $rows = $rows->groupBy('label_tree_id');
+            $trees = LabelTree::whereIn('id', $rows->keys())->pluck('name', 'id');
+
+            foreach ($trees as $id => $name) {
+                $csv = $this->createCsv($rows->get($id));
+                $this->tmpFiles[] = $csv;
+                $toZip[$csv->getPath()] = $this->sanitizeFilename("{$id}-{$name}", 'json');
+            }
+        } elseif ($this->shouldSeparateUsers() && $rows->isNotEmpty()) {
+            $rows = $rows->groupBy('user_id');
+            $users = User::whereIn('id', $rows->keys())
+                ->selectRaw("id, concat(firstname, ' ', lastname) as name")
+                ->pluck('name', 'id');
+
+            foreach ($users as $id => $name) {
+                $csv = $this->createCsv($rows->get($id));
+                $this->tmpFiles[] = $csv;
+                $toZip[$csv->getPath()] = $this->sanitizeFilename("{$id}-{$name}", 'json');
+            }
+        } else {
+            $csv = $this->createCsv($rows);
+            $this->tmpFiles[] = $csv;
+            $toZip[$csv->getPath()] = $this->sanitizeFilename("{$this->source->id}-{$this->source->name}", 'json');
+        }
+        $this->executeScript('toCoco', $path);
```

Comment:
You could add a comment here that `$path` is not really used here and instead, the temporary CSV files are overwritten with the JSON COCO reports.
```suggestion
        $this->executeScript('toCoco', '');
```
If you use `''` as second argument as suggested above, you could probably also do this in the Python script (i.e. start at `2` instead of `3`):
```python
paths = sys.argv[2:]
```